src/test/groovy/com/commercehub/gradle/plugin/avro/OptionsFunctionalSpec.groovy

```diff
@@ -281,6 +283,7 @@ class OptionsFunctionalSpec extends FunctionalSpec {
         result.output.contains("Invalid default for field name: null not a \"string\"")
     }
 
+    @Ignore("1.9 compiler validates defaults and fails")
```

Comment:
I've added this ignore because this test is now failing. It looks like maybe the compiler now does a validation it was not doing before? I didn't see a way to pass this option to the compiler.

Reply:
It looks like in 1.9, they've changed behavior.

https://issues.apache.org/jira/browse/AVRO-2035

For Avro 1.9+, I don't think that `validateDefaults == false` will be a setting that makes sense any more.  If we can support 1.9 without it otherwise being a breaking change, I think my preference will be to deprecate this setting (logging a warning if it's set to a non-true value), and remove it in the next version of the plugin that breaks compatibility.  Otherwise, we'll just remove it completely now.

Reply:
Makes sense. Can we make the test conditional on Avro <= 1.8 somehow?

Reply:
We could in theory, but parsing the Avro version in such a way to correctly do the comparison is a bit more effort than it's worth.  Just leave this `@Ignore`'ed and add a TODO comment in here for me to clean up.

Reply:
Given our findings in other comments that 1.9 is a breaking change, I'll go ahead and remove this completely now. Should we keep the setting, with no-op behavior and a deprecated notice so we don't break builds, or just remove it completely?

Reply:
My preferred approach is to add a note to the changelog that the setting was removed, and remove both the setting and the test coverage completely.  As you say, if someone has explicitly set this configuration setting, they'll get an error that the field no longer exists when they upgrade the plugin, and need to remove it to fix the error.

If the upstream project took better care to not break compatibility as often as they do, I'd be more inclined towards a more conservative stance.

Reply:
I guess that's why its a major 1.9 release. I suppose its ok to break compatibility given that. The issue you referenced actually mentions that, and postpones the change until 1.9 accordingly, so it does seem they thought this through.

That being said, the 1.9 parser still takes the `setValidateDefaults` parameter, which is definitely weird, since its unclear to me now whether this still does something or not. Perhaps its just this particular test case that is no longer affected by this option, but this option may still affect other behavior?

If that is not the case, then we should probably just advocate upstream to remove this setting as well, before the release of 1.9.

Thoughts?

Reply:
FYI: https://issues.apache.org/jira/browse/AVRO-2366

Reply:
As I noted in the issue above, this sounds like an implementation of Postel's law in Avro -- always produce valid schemas, but accept invalid ones. With that in mind, it makes sense to remove this option. I'll submit another commit with the option removed.

Reply:
I'll be interested in what they say in response to that issue.  As the Avro setting is on the parser, and the broken behavior is in the compiler, I could see there potentially being an argument that the setting isn't intended to be used in that way.